ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -16,21 +16,22 @@ def initialize(repository:,
     end
 
     def publish_events(events, stream_name: GLOBAL_STREAM, expected_version: :any)
-      append_to_stream(events, stream_name: stream_name, expected_version: expected_version)
-      events.each do |ev|
-        event_broker.notify_subscribers(ev)
+      prepared_events = prepare_events(events)
+      serialized_events = serialize_events(prepared_events)
+      append_to_stream_serialized_events(serialized_events, stream_name: stream_name, expected_version: expected_version)
+      prepared_events.zip(serialized_events) do |payload|
+        event_broker.notify_subscribers(payload)
       end
       :ok
     end
 
     def publish_event(event, stream_name: GLOBAL_STREAM, expected_version: :any)
-      publish_events([event], stream_name: stream_name, expected_version: expected_version)
+      publish_events(event, stream_name: stream_name, expected_version: expected_version)
```

Comment:
seems weird to me we are not passing `[event]` . but not worth much discussion considering #366